lizmap/modules/lizmap/classes/lizmapRepository.class.php

```diff
@@ -125,6 +125,42 @@ public function update($data)
         return $modified;
     }
 
+    /**
```

Comment:
I would have modified the existing function in https://github.com/drillinP/lizmap-web-client/blob/backport-2199-to-release_3_3/lizmap/modules/lizmap/classes/lizmap.class.php#L280
instead of recreating a function here. 

Be careful to modify the function by just adding the new param and using it with `if ($keepReference) {...`

In 3.4, this function has been moved in the lizmapRepository class, but not in 3.3